[Liew Jun Wei] Duke Increments

bigjunnn opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

bigjunnn committed:
Added greet, echo and exit function

bigjunnn committed:
Added support for adding to list & listing items

bigjunnn committed:
Added support for mark as done
- Done is marked as X
- Undone is unmarked

bigjunnn committed:
Added support for different task type:
- ToDo
- DeadLines
- Events

bigjunnn committed:
Added support for error handling:
- Description of todo cannot be empty
- Unexpected commands
- Done should be followed by integer, and within range of list
- Deadlines should be followed with /by
- Events should be followed with /at

bigjunnn committed:
Added support for delete function & error handling:
- Delete should be followed by an integer, within range of list

bigjunnn committed:
Merge branch 'gradle'

bigjunnn committed:
Configured gradle with the main application name: "Duke"

bigjunnn committed:
Branch-7 commit - Added support for saving & loading tasks

bigjunnn committed:
Added support for DateTime conversion:

bigjunnn committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Branch-7 commit - Added support for saving & loading tasks

bigjunnn committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Added support for DateTime conversion:

# Conflicts:
#	src/main/java/Duke.class
#	src/main/java/Duke.java

bigjunnn committed:
Added support for level 8:

bigjunnn committed:
Minor bug fix:
- Date formatting when loading Duke

bigjunnn committed:
Added support for errors:
- Date should be supplied in dd/mm/yy, followed by time in 24h format

bigjunnn committed:
Added more OOP elements:
- Added Storage class to handle files
- Added Parser class to handle reading commands
- Added taskList to manage tasks
- Added Ui to handle printing welcome message

bigjunnn committed:
- Fixed exception formatting
- Fixed done / delete bug

bigjunnn committed:
JUnit support:
- Added test cases for ToDo
- Added test cases for DeadLine
- Added test cases for EventTest

bigjunnn committed:
A-Jar
- Added jar support with gradle

bigjunnn committed:
JavaDocs
- Added JavaDocs comment for classes & methods

bigjunnn committed:
CodingStandard

bigjunnn committed:
Level-9
- Added support for find function

bigjunnn committed:
Merge branch 'branch-A-CodingStandard'

bigjunnn committed:
Merged Level-9 to master

bigjunnn committed:
Refactored Code

bigjunnn committed:
Checkstyle Support:
- Added checkstyle support and fixed checkstyle errors

ambhinav commented on src/main/java/Task.java:
```diff
@@ -1,15 +1,24 @@
+    return "[" + this.getStatusIcon() + "] " + this.description;
+  }
+
+  public String getStatusIcon() {
```
good practice to abstract the status icon +1

ambhinav commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,238 @@
+      Task removed = tasks.get(taskNum);
+      tasks.remove(taskNum);
+
+      System.out.println("\t____________________________________________________________");
```
I think it would have been better to abstract adding the borders to a separate method so as to reduce repeated code. At least that is how i did it.

ambhinav commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,46 @@
+        Parser p = new Parser(command, description);
+        p.executeOnly(this);
+      }
+    } catch (FileNotFoundException e) {
```
Perhaps you could create the file if it is not found. I thought that would be better as the user does'nt need to go create the file later

ambhinav commented on src/main/java/Task.java:
```diff
@@ -9,6 +13,9 @@ public Task(String description) {
   }
 
+  /**
+   * Sets the task to be done.
```
i think u meant "Set the task as done" instead?

ambhinav commented:
Hey Jun Wei, I think your code was very easy to follow, especially since you documented it well. Also you commited frequently and with clear messages which is something I need to learn from u lol

- I think you made good use of all the OOP concepts. You managed to extract all the necessary classes and made the main method look clean.

- Your adherence to coding standard was great and it made it very easy to read the code

Improvments I thought could be made:

- Creating an abstract commands class and separate commands that can extend from it. (At least that's how I did it) So that later on if you want to add more commands you don't need to repeat common code

- The rest i wrote as in-line comments


Thanks

bigjunnn committed:
Added support for GUI:
- Provided image for user and duke
- Display input in dialog boxes

bigjunnn committed:
Added packaging support

bigjunnn committed:
Level-10
- Added packaging
- GUI feature added

bigjunnn committed:
Add assertions

bigjunnn committed:
Improve code quality
Remove unused code.

Change some 'if' statements to 'switch'.

Use string builders for string concatenation.

bigjunnn committed:
Add help function

bigjunnn committed:
Merge pull request #3 from bigjunnn/branch-A-Assertions

Add assertions

bigjunnn committed:
Merge branch 'master' into branch-A-CodeQuality

bigjunnn committed:
Merge pull request #4 from bigjunnn/branch-A-CodeQuality

Improve code quality

bigjunnn committed:
Merge branch 'master' into C-Help

bigjunnn committed:
Merge pull request #5 from bigjunnn/C-Help

Add help function

bigjunnn committed:
Fix minor bugs

bigjunnn committed:
Add travis support

bigjunnn committed:
Fix checkstyle errors

bigjunnn committed:
Fix bugs

bigjunnn committed:
Update Parser.java

bigjunnn committed:
Update DeadLineTest.java

bigjunnn committed:
Update EventTest.java

bigjunnn committed:
Update ToDoTest.java

bigjunnn committed:
Update EventTest.java

bigjunnn committed:
Update DeadLineTest.java

bigjunnn committed:
Merge pull request #6 from bigjunnn/A-Travis

Add travis support

TanZhanMing commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,102 @@
+            ToDo t = new ToDo(taskDescription);
+            tasks.addTask(t);
+
+            // Saving to text file
```
I really like how there are comments within the methods themselves instead of just dumping everything in the javadoc comment up top. Makes the code easier to understand and aids in debugging.

TanZhanMing commented on src/main/java/utils/Storage.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public void deleteTask(int index) throws IOException {
+        int count = 0;
+        StringBuilder finalText = new StringBuilder();
```
From what I can gather (please correct me if I am wrong), you are trying to gather all the data from the existing text file and store them in the variable 'finalText' and then override the existing data when there are changes.
I feel that there might be pros and cons to this.

Pros: saves space and computational power as there is no need to create a separate temp.txt file and then replace the tasks.txt file every time there is an update
Cons: On the off chance the data set from tasks.txt is too big (2^32-1), the 'finalText' String might not be able to hold them.

TanZhanMing reviewed (commented):
Hello Jun Wei,

First of all, I would like to apologise as I had actually commented last week but at the wrong place. I commented at https://github.com/nus-cs2103-AY1920S1/duke/pull/194 instead of here and thus I just copied everything I posted there here. Once again, I am truly sorry for my late review and feedback.

Your code was very nicely written and it shows how much you value clarity in your work. There were also quite a number of methods that you used which I could definitely pick up on to improve my own code base.

The good points:

Really easy to understand code. Your code was very modular in nature which aids in comprehension as I dislike reading long blocks of code. Good usage of white-space between lines, comments within methods to keep readers up to speed etc.
Well-implemented methods. Took me a while to figure out what you were trying to do with the 'getSuffix' method but once I did, I found it to be really creative. Other methods were also clear and concise.
Points for improvement:

Maybe try and utilize switch-case instead of if-else blocks for some of the methods. It can make the code look neater.
Maybe intentionally leave a few mistakes here and there so reviewers don't have to look so hard for points for improvements? I'm just kidding.
All in all, really impressive on your part; I feel like I have learned quite alot from this.

bigjunnn committed:
Add User Guide

bigjunnn committed:
Add images to user guide

bigjunnn committed:
Edit user guide

bigjunnn committed:
Set theme jekyll-theme-tactile

bigjunnn committed:
Rename files and optimise imports

bigjunnn committed:
Edit image links for user guide

bigjunnn committed:
Fix images for user guide

bigjunnn committed:
Fix broken links in user guide

bigjunnn committed:
Fix jar files for release

bigjunnn committed:
Add cross-platform support for jar files

bigjunnn committed:
Make minor edits to user guide

bigjunnn committed:
Fix checkstyle errors

bigjunnn committed:
Fix GUI to allow re-sizing of screen

bigjunnn committed:
Fix Code Quality

bigjunnn committed:
Set theme jekyll-theme-modernist
